### PR TITLE
[Frontend] Implement Bookmark Page

### LINKED ISSUE #594: [Frontend] Implement Bookmark Page
# 📌 Implement a Feature

## 🏡 Component 
- [x] Frontend

## Description

Implement Bookmark Page. Show bookmarked Questions in the Bookmark Page.

## 📋 Acceptance Criteria
- [X] The Bookmarks button can be reached from the User's Profile.
- [X] The Boomarks button forwards to page Bookmarks.
- [X] Boomarks page shows all bookmarked questions.

## 📃  Related Requirements

Insert bookmarks to the requirements document.

## 👁‍🗨 Reviewer
@mmtftr

### PR DESCRIPTION
**Pull request by ozdentarikcan**
## 📋 Proposed Changes

  - The bookmark page is implemented.
  - The button that navigates to bookmark page is implemented inside the user profile page.

## Related Issue
Closes #594 

## 🧪 Included Tests
- [X] Bookmarked questions are rendered correctly.

### CODE REVIEW

File: frontend/src/routes/profile.tsx [hunk 1]
```diff
@@ -153,9 +153,14 @@ export default function Profile() {
                 {isPending ? "Saving..." : "Save"}
               </Button>
             ) : (
-              <Button onClick={() => setEditing(true)} variant="outline">
-                Edit profile
-              </Button>
+              <div style={{ display: 'flex', gap: '15px', justifyContent: 'center', alignItems: 'center' }}>
```
Comment (mmtftr) [inline frontend/src/routes/profile.tsx:156]: tailwind classes pls: className="flex gap-4 justify-center items-center"

File: frontend/src/routes/index.tsx [hunk 2]
```diff
@@ -26,6 +27,10 @@ export const routes: RouteObject[] = [
     path: "/login",
     Component: Login,
   },
+  {
+    path: "/bookmarkedquestions",
```
Comment (mmtftr) [inline frontend/src/routes/index.tsx:31]: nit: I prefer /bookmarks

File: frontend/src/routes/bookmarked.tsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  const next = () => {
+    setPageSize(pageSize + 20);
+  };
+
+  return (
+    <div className="container flex flex-col gap-2 py-8">
+      <div className="flex items-center justify-between">
+        <h1 className="text-2xl font-bold ">
+          {questions.length
+            ? `Last ${resultListData.totalItems} bookmarked questions shown.`
```
Comment (mmtftr) [inline frontend/src/routes/bookmarked.tsx:50]: `totalItems` is not the # of shown items but the total items.

Also, in terms of UX, it doesn't help the user to know they're seeing X items without knowing how many more there are. Instead, let's just tell them the total number of items here. `Bookmarks (${...totalItems})`

File: frontend/src/routes/bookmarked.test.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        error: null,
+        data: {
+          data: { items: mockQuestions, totalItems: mockQuestions.length },
+        },
+      } as QueryObserverSuccessResult<unknown, GetBookmarkedQuestionsError>);
+  
+      render(
+        <MemoryRouter initialEntries={["/bookmarked"]}>
+          <Routes>
+            <Route path="/bookmarked" element={<BookmarkedQuestions />} />
```
Comment (mmtftr) [inline frontend/src/routes/bookmarked.test.tsx:93]: test route != actual route here

**Review by mmtftr — CHANGES_REQUESTED** [review]
Thanks a lot for the work here, I just have a couple of style and minor comments

File: frontend/src/routes/profile.tsx (hunk 1 above)
Comment (ozdentarikcan) [reply frontend/src/routes/profile.tsx]: Done.

File: frontend/src/routes/index.tsx (hunk 2 above)
Comment (ozdentarikcan) [reply frontend/src/routes/index.tsx]: Done. Updated file names to bookmarks as well.

File: frontend/src/routes/bookmarked.tsx (hunk 3 above)
Comment (ozdentarikcan) [reply frontend/src/routes/bookmarked.tsx]: Done.

File: frontend/src/routes/bookmarked.test.tsx (hunk 4 above)
Comment (ozdentarikcan) [reply frontend/src/routes/bookmarked.test.tsx]: Updated the test route to /bookmarks with the renaming of the routes.

**Comment by ozdentarikcan** [comment]
Thanks for the feedback! I updated all the files accordingly.

**Review by mmtftr — APPROVED** [review]
Thanks a lot, looks good to me

**Pull request merged**